airflow/contrib/operators/mysql_to_gcs.py

```diff
@@ -0,0 +1,183 @@
+        :param approx_max_file_size_bytes: This operator supports the ability
+            to split large table dumps into multiple files (see notes in the
+            filenamed param docs above). Google cloud storage allows for files
+            to be a maximum of 4GB. This param allows developers to specify the
+            file size of the splits.
+        :type approx_max_file_size_bytes: long
+        :param mysql_conn_id: Reference to a specific MySQL hook.
+        :type mysql_conn_id: string
+        :param google_cloud_storage_conn_id: Reference to a specific Google
+            cloud storage hook.
+        :type google_cloud_storage_conn_id: string
+        :param delegate_to: The account to impersonate, if any. For this to
+            work, the service account making the request must have domain-wide
+            delegation enabled.
+        """
+        super(MySqlToGoogleCloudStorageOperator, self).__init__(*args, **kwargs)
+        self.sql = sql;
+        self.bucket = bucket
+        self.filename = filename
+        self.schema_filename = schema_filename
+        self.approx_max_file_size_bytes = approx_max_file_size_bytes
+        self.mysql_conn_id = mysql_conn_id
+        self.google_cloud_storage_conn_id = google_cloud_storage_conn_id
+        self.delegate_to = delegate_to
+
+    def execute(self, context):
+        cursor = self._query_mysql()
+        files_to_upload = self._write_local_data_files(cursor)
+
+        # If a schema is set, create a BQ schema JSON file.
+        if self.schema_filename:
+            files_to_upload.update(self._write_local_schema_file(cursor))
+
+        # Flush all files before uploading
+        for file_handle in files_to_upload.values():
+            file_handle.flush()
+
+        self._upload_to_gcs(files_to_upload)
+
+        # Close all temp file handles.
+        for file_handle in files_to_upload.values():
+            file_handle.close()
+
+    def _query_mysql(self):
+        """
+        Queries mysql and returns a cursor to the results.
+        """
+        mysql = MySqlHook(mysql_conn_id=self.mysql_conn_id)
+        conn = mysql.get_conn()
+        cursor = conn.cursor()
+        cursor.execute(self.sql)
+        return cursor
+
+    def _write_local_data_files(self, cursor):
+        """
+        Takes a cursor, and writes results to a local file.
+
+        :return: A dictionary where keys are filenames to be used as object
+            names in GCS, and values are file handles to local files that
+            contain the data for the GCS objects.
+        """
+        schema = map(lambda schema_tuple: schema_tuple[0], cursor.description)
+        file_no = 0
+        tmp_file_handle = NamedTemporaryFile(delete=True)
+        tmp_file_handles = { self.filename.format(file_no): tmp_file_handle }
+
+        for row in cursor:
+            row_dict = dict(zip(schema, row))
+            # TODO validate that row isn't > 2MB. BQ enforces a hard row size of 2MB.
```

Comment:
I plan to do it in next PR. I plan to hit this problem when we pull some of our tables, and what to fix it when I have a real issue with it. I am just at the point where I have enough back-logged non-merged code that I want to split the commits up.